metricflow/dataflow/optimizer/predicate_pushdown_optimizer.py

```diff
@@ -56,20 +55,45 @@ def track_pushdown_state(self, pushdown_state: PredicatePushdownState) -> Iterat
         This retains a sequence of pushdown state objects to allow for tracking pushdown opportunities along
         the current branch. Each entry represents the predicate pushdown state as of that point, and as such
         callers need only concern themselves with the value returned by the last_pushdown_state property.
+
+        The back-propagation of pushdown_applied_where_filter_specs is necessary to ensure the outer query
+        node can evaluate which where filter specs needs to be applied. We capture the complete set because
+        we may have sequenced nodes where some where filters are applied (e.g., time dimension filters might
+        be applied to metric time nodes, etc.).
         """
         self._current_branch_state.append(pushdown_state)
         yield
-        self._current_branch_state.pop(-1)
+        last_visited_pushdown_state = self._current_branch_state.pop(-1)
+        if len(last_visited_pushdown_state.applied_where_filter_specs) > 0:
+            pushdown_applied_where_filter_specs = frozenset.union(
+                *[
+                    last_visited_pushdown_state.applied_where_filter_specs,
+                    self.last_pushdown_state.applied_where_filter_specs,
+                ]
+            )
+            self.override_last_pushdown_state(
+                PredicatePushdownState.with_pushdown_applied_where_filter_specs(
+                    original_pushdown_state=self.last_pushdown_state,
+                    pushdown_applied_where_filter_specs=pushdown_applied_where_filter_specs,
+                )
+            )
 
     @property
     def last_pushdown_state(self) -> PredicatePushdownState:
         """Returns the last seen PredicatePushdownState.
 
-        This is nearly always the input state a given node processing method should be using for pushdown operations.
+        This is the input state a given node processing method should be using for pushdown operations.
         """
-        if len(self._current_branch_state) > 0:
-            return self._current_branch_state[-1]
-        return self._initial_state
+        return self._current_branch_state[-1]
```

Comment:
No, because there's always a value set since we initialize it non-empty, but I should put in an assertion guard so we get a useful error message if anybody changes that.